### ISSUE TITLE
3D plot fails with `IndexError`

### ISSUE THREAD

**Issue by wtbarnes** [sunpy/ndcube#215]
### Description

I have a 3D AIA data cube (space, space, time). I would like to animate this data cube by plotting sequential 2D images in time. I tried doing this by simply calling the `.plot` method on the data cube. However, this yields a lengthy traceback and an `IndexError`.

### Expected behavior
A 2D animated plot of my image in time

### Actual behavior
```python-traceback
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-4-7537c79d1eb7> in <module>
----> 1 cube.plot()

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/ndcube/mixins/plotting.py in plot(self, axes, plot_axis_indices, axes_coordinates, axes_units, data_unit, **kwargs)
     93                     ax = self._plot_3D_cube(
     94                         plot_axis_indices=plot_axis_indices, axes_coordinates=axes_coordinates,
---> 95                         axes_units=axes_units, **kwargs)
     96         return ax
     97 

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/ndcube/mixins/plotting.py in _plot_3D_cube(self, plot_axis_indices, axes_coordinates, axes_units, data_unit, **kwargs)
    331         # using NDCube's wcs object.
    332         new_axes_coordinates, new_axes_units, default_labels = \
--> 333             self._derive_axes_coordinates(axes_coordinates, axes_units, data.shape, edges=True)
    334 
    335         if (axes_coordinates[plot_axis_indices[0]] is None and

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/ndcube/mixins/plotting.py in _derive_axes_coordinates(self, axes_coordinates, axes_units, data_shape, edges)
    458                     index = np.delete(index, reduce_axis)
    459                     # Reduce the data by taking mean
--> 460                     new_axis_coordinate = np.mean(new_axis_coordinate, axis=tuple(index))
    461 
    462             elif isinstance(axis_coordinate, str):

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/numpy/core/fromnumeric.py in mean(a, axis, dtype, out, keepdims)
   3113             pass
   3114         else:
-> 3115             return mean(axis=axis, dtype=dtype, out=out, **kwargs)
   3116 
   3117     return _methods._mean(a, axis=axis, dtype=dtype,

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/astropy/units/quantity.py in mean(self, axis, dtype, out)
   1520 
   1521     def mean(self, axis=None, dtype=None, out=None):
-> 1522         return self._wrap_function(np.mean, axis, dtype, out=out)
   1523 
   1524     def round(self, decimals=0, out=None):

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/astropy/units/quantity.py in _wrap_function(self, function, unit, out, *args, **kwargs)
   1500             kwargs['out'] = check_output(out, unit, arrays, function=function)
   1501         # Apply the function and turn it back into a Quantity.
-> 1502         result = function(*args, **kwargs)
   1503         return self._result_as_quantity(result, unit, out)
   1504 

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/numpy/core/fromnumeric.py in mean(a, axis, dtype, out, keepdims)
   3116 
   3117     return _methods._mean(a, axis=axis, dtype=dtype,
-> 3118                           out=out, **kwargs)
   3119 
   3120 

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/numpy/core/_methods.py in _mean(a, axis, dtype, out, keepdims)
     60 
     61     is_float16_result = False
---> 62     rcount = _count_reduce_items(arr, axis)
     63     # Make this warning show up first
     64     if rcount == 0:

~/anaconda3/envs/aia-on-pleiades/lib/python3.7/site-packages/numpy/core/_methods.py in _count_reduce_items(arr, axis)
     53     items = 1
     54     for ax in axis:
---> 55         items *= arr.shape[ax]
     56     return items
     57 

IndexError: tuple index out of range
```

### Steps to Reproduce

I apologize ahead of time that this MWE is not very minimal. **I attempted to reproduce this error using the dummy example in the docs, but could not do so.** I will update this issue if I can find a more simple case that reproduces this issue. Note that I've experienced this same issue with other similarly shaped data cubes from different data sets

1. Download an AIA data cube that has been aligned with the IRIS FOV [here](www.lmsal.com/solarsoft/irisa/data/level2_compressed/2018/05/29/20180529_183737_3600104031/iris_l2_20180529_183737_3600104031_SDO.tar.gz)
2. Untar the directory
3. Run the following script (in the same directory that you untarred the data into)

```python
import astropy.wcs
import astropy.io.fits
from sunpy.util import MetaDict
import ndcube
filename = 'iris_l2_20180529_183737_3600104031_SDO/aia_l2_20180529_183737_3600104031_171.fits'
with astropy.io.fits.open(filename) as hdul:
    data = hdul[0].data
    header = hdul[0].header
meta = MetaDict(dict(header))
del meta['history']
del meta['comment']
wcs = astropy.wcs.WCS(meta)
cube = ndcube.NDCube(data, wcs, meta=meta)
cube.plot()
```

### System Details

  - ndcube Version: 1.2.1
  - SunPy  Version: 1.0.3
  - astropy Version: 3.2.2
  - Python Version: 3.7.3
  - OS information: macOS 10.14.5 (Mojave)

**Comment by DanRyanIrish** [comment]
Hi @wtbarnes.  I tried reproducing this but fell at the first hurdle.  The url to the aligned AIA cube could not be found...

**Comment by wtbarnes** [comment]
Ok for some odd reason clicking on the link gives a 404, but if you manually paste `www.lmsal.com/solarsoft/irisa/data/level2_compressed/2018/05/29/20180529_183737_3600104031/iris_l2_20180529_183737_3600104031_SDO.tar.gz` into the address bar of your web browser, it should prompt you to download the tar file, at least on Firefox.

**Comment by wtbarnes** [comment]
If it helps, I also played around with this same data set in NDCube in [this gist](https://gist.github.com/wtbarnes/3890a845882a9423b2c577f93453fcb4). The slicing seems to work quite nicely!

**Comment by DanRyanIrish** [comment]
@wtbarnes is this still an issue with ndcube v1.3 and/or master?

**Comment by jacobdparker** [comment]
I have been having similar struggles with 3D plots @wtbarnes and @DanRyanIrish . Attached is code that reproduces a similar (if not the same) error with an NDCube modeled after the example here: https://docs.sunpy.org/projects/ndcube/en/stable/ndcube.html in `cube.plot()`.  When following the example exactly I was able to get a 3D plot, but with the latitude and longtitude axes swapped (labeled incorrectly) in the plot.  2D plotting of a sliced cube plots as expected.

In the attached notebook I found that using `sunpy.visualization.animator.ArrayAnimatorWCS` created a functional and properly labeled 3D plot for a variety of 'x' and 'y' choices. 

 Is it time to replace the deprecated `ImageAnimatorWCS` in the _plot_3D_cube?

I am currently running:
NDCube version 1.3.2
Sunpy version 1.1.1
Astropy version 4.0



```
import numpy as np  
import matplotlib.pyplot as plt
import ndcube
import astropy.wcs
import sunpy`

data = np.ones([50,1024,2048])

wcs_input_dict = {
'CTYPE1': 'HPLN-TAN', 'CUNIT1': 'deg', 'CDELT1': 1/3600, 'CRPIX1': 1024, 'CRVAL1': 0, 'NAXIS1': 2048,
'CTYPE2': 'HPLT-TAN','CUNIT2': 'deg', 'CDELT2': 1/3600, 'CRPIX2': 512, 'CRVAL2': 0, 'NAXIS2': 1024,
'CTYPE3': 'WAVE    ', 'CUNIT3': 'Angstrom', 'CDELT3': 1, 'CRPIX3': 25, 'CRVAL3': 0, 'NAXIS3': 50}
input_wcs = astropy.wcs.WCS(wcs_input_dict)

cube = ndcube.NDCube(data,input_wcs)
print(cube.wcs)

sliced_cube = cube[0]
sliced_cube.plot()

sunpy.visualization.animator.ArrayAnimatorWCS(cube.data,cube.wcs,slices = ('x','y',0))

plt.figure()
cube.plot(image_axis = [0,1])

plt.show()
```

**Comment by wtbarnes** [comment]
Sorry, realized I never followed up on this. Yes, this is still an issue on 1.3.2 for me.

**Comment by DanRyanIrish** [comment]
Hi @jacobdparker, thanks for you comment.  For v2.0 we are working on a massive refactor that should resolve this issue, if it is not already resolved by changes we've made in 1.4.1.
